posthog/temporal/tests/utils/workflow.py

```diff
@@ -92,26 +104,33 @@ async def run(self, inputs: WaitInputs) -> None:
         if inputs.mode == WaitMode.ASYNC:
             await workflow.execute_activity_method(
                 Waiter.wait_for_activity,
-                inputs.wait_for,
+                WaitInputsActivity(wait_for=inputs.wait_for, raise_on_shutdown=inputs.raise_on_shutdown),
                 # Setting a timeout is required.
-                start_to_close_timeout=dt.timedelta(minutes=1),
-                heartbeat_timeout=dt.timedelta(seconds=5),
+                start_to_close_timeout=dt.timedelta(seconds=inputs.start_to_close_timeout_seconds),
+                heartbeat_timeout=dt.timedelta(seconds=inputs.heartbeat_timeout_seconds),
                 retry_policy=RetryPolicy(
                     initial_interval=dt.timedelta(seconds=1),
                     maximum_interval=dt.timedelta(seconds=1),
-                    maximum_attempts=1,
+                    maximum_attempts=inputs.maximum_attempts,
                 ),
             )
         elif inputs.mode == WaitMode.SYNC:
             await workflow.execute_activity_method(
                 Waiter.wait_for_activity_sync,
-                inputs.wait_for,
+                WaitInputsActivity(wait_for=inputs.wait_for, raise_on_shutdown=inputs.raise_on_shutdown),
                 # Setting a timeout is required.
-                start_to_close_timeout=dt.timedelta(minutes=1),
-                heartbeat_timeout=dt.timedelta(seconds=10),
+                start_to_close_timeout=dt.timedelta(seconds=inputs.start_to_close_timeout_seconds),
+                heartbeat_timeout=dt.timedelta(seconds=inputs.heartbeat_timeout_seconds),
                 retry_policy=RetryPolicy(
                     initial_interval=dt.timedelta(seconds=1),
                     maximum_interval=dt.timedelta(seconds=1),
-                    maximum_attempts=1,
+                    maximum_attempts=inputs.maximum_attempts,
                 ),
             )
+
+
+WAITER = Waiter()
```

Comment:
how will this global class behave when multiple instances of the same workflow are running on the same worker for instance?

Reply:
They will all have a reference to the same waiter. Shouldn't matter much as I am not planning on accessing it or running multiple instances of this workflow. It's used in tests but we do need an instance to pass to the worker.